Source/WTF/wtf/PlatformUse.h

```diff
@@ -348,9 +348,19 @@
 #endif
 
 #if !defined(USE_LIBPAS_JIT_HEAP) && !USE(SYSTEM_MALLOC)
-#include <bmalloc/BPlatform.h>
-#if BENABLE(LIBPAS) && OS(DARWIN)
+// Inlined `BENABLE(LIBPAS) && OS(DARWIN)`, to avoid a header dependency on bmalloc.
+#if __SIZEOF_POINTER__ == 8 && OS(DARWIN)
 #define USE_LIBPAS_JIT_HEAP 1
+
+// If bmalloc *is* available (typical outside of production build contexts), check that the inlined
+// definition matches.
+#if defined(__has_include) && __has_include(<bmalloc/BPlatform.h>)
+#include <bmalloc/BPlatform.h>
+#if USE_LIBPAS_JIT_HEAP != BENABLE(LIBPAS)
+#error "Inlined BENABLE(LIBPAS) in " #file " does not match actual BENABLE(LIBPAS)"
+#endif
+#endif
```

Comment:
This might be overkill, since I don't think the availability of libpas is subject to change. wdyt?

Reply:
This seems quite fragile, the reason for all this is to not duplicate the logic for JIT heap, which did and will change... I don't really see a good solution that wouldn't involve unifdef.

Reply:
> I don't really see a good solution that wouldn't involve unifdef.

What do you have in mind? unifdef will never delete an `include` statement, so I don't see how any amount of postprocessing could get rid of this header dependency.

Reply:
Right, we would need to provide the value to unifdef in some other way.